src/rgw/rgw_op.cc

```diff
@@ -1051,12 +1051,17 @@ int handle_cloudtier_obj(req_state* s, const DoutPrefixProvider *dpp, rgw::sal::
         s->err.message = "restore is still in progress";
       }
       return op_ret;
-    } else if ((!restore_op) && (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress)) {
-      op_ret = -ERR_REQUEST_TIMEOUT;
-      ldpp_dout(dpp, 5) << "restore is still in progress, please check restore status and retry" << dendl;
-      s->err.message = "restore is still in progress";
-    } else { // CloudRestored..return success
-      return 0;
+    } else if (restore_status == rgw::sal::RGWRestoreStatus::RestoreAlreadyInProgress) {
```

Comment:
Can you copy all the changes from https://github.com/ceph/ceph/commit/1c2b0c7ff162b830fd3092e707bb747b84c39c96 (including comments)

Reply:
Copied.